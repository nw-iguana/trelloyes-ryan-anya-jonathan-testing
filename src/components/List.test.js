import React from 'react';
import ReactDOM from 'react-dom';
import List from './List.js';
import Store from '../store.js';
import renderer from 'react-test-renderer';



describe('test from displaying card component', ()=> {
    it('renders a list without crashing', ()=> {
        const div = document.createElement('div');
        ReactDOM.render(<List key={{id: "1"}}
                              cards={["b", "c", "d", "f", "h", "i", "k"]}
                              header={"Second list"}
                              store={Store}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it('renders the UI as expected', ()=> {
        const tree = renderer
            .create(<List key={{id: "1"}}
                          cards={["b", "c", "d", "f", "h", "i", "k"]}
                          header={"Second list"}
                          store={Store}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})
