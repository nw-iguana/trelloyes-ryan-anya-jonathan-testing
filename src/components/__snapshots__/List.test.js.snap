// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test from displaying card component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Second list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
         
        Second card
         
      </h3>
      <p>
         
        lorem ipsum
         
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
         
        Third card
         
      </h3>
      <p>
         
        lorem ipsum
         
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
         
        Fourth card
         
      </h3>
      <p>
         
        lorem ipsum
         
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
         
        Sixth card
         
      </h3>
      <p>
         
        lorem ipsum
         
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
         
        Eighth card
         
      </h3>
      <p>
         
        lorem ipsum
         
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
         
        Ninth card
         
      </h3>
      <p>
         
        lorem ipsum
         
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
         
        Eleventh card
         
      </h3>
      <p>
         
        lorem ipsum
         
      </p>
    </div>
  </div>
</section>
`;
